### DISCUSSION TITLE
knit: yaml equivalent in quarto

### DISCUSSION THREAD

**Discussion by yonicd** [quarto-dev/quarto-cli#7789] [Q&A]
### Description

what is the equivalent in the yaml in quarto for what `knit:` does in rmarkdown

eg

if my rmd is a template and it is called index.rmd and i want to output it to another directory in the tree structure under another name.

```
--- 
title: "Title of Report"
output: 
  pdf_document:
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_dir = '../../../reports', output_file = 'report.pdf') })
---
```

qmd ignores the knit param

```
---
title: "Title of Report"
format: pdf
knit: (function(inputFile) {
  quarto::quarto_render(input = inputFile, output_file = 'test.pdf') })
```

**Comment by mcanouil** [comment]
There is no equivalent, because Quarto is a command line interface not an R package contrary to knitr/rmarkdown.
Here the "quarto" R package calls Quarto CLI from within R, but RStudio and VSCode don't do that, they run directly Quarto CLI.
Quarto calls knitr, not the other way around contrary to knitr/rmarkdown.
From `quarto render --help`, you can see there is a parameter to set the output directory for Quarto project and the output file name.

Did you read/go through the documentation, especially the starting guides?
It is documented that for project you can set the `output-dir` in `_quarto.yml`.

```yaml
project:
  output-dir: directory
```

Or to change the filename, there is `output-file` documented in every reference pages.


Also, the extension for Quarto documents is "qmd" not "rmd", even if it matters little compared to using the proper option inside such as `format` and not `output`.
Better to make it clear by using the proper extension and make it consistent with its content.

**Comment by yonicd** [comment]
thanks for the response.

My main issue is the renaming of the output file different from the original input file.

**Reply by mcanouil** [reply]
As I said, this is as simple as setting `output-file: whatever.html` in the frontmatter.

**Comment by yonicd** [comment]
ok. i see now. the options in the cli are mimicked in the yml header.

```
---
title: "Exhibit A"
format:  pdf
output-file: test.pdf
---
```

thanks for the help

**Reply by mcanouil** [reply]
You should take a look at the reference pages of the documentation.
This is all written and searchable via the search engine.

**Comment by yonicd** [comment]
actually the cli help shows a different input, which threw me off in the first place. there is no `output-file` it is `output`

```
quarto render --help

  Usage:   quarto render [input] [args...]
  Version: 1.3.433

  Description:

    Render files or projects to various document types.

  Options:

    -h, --help                          - Show this help.
    -t, --to                            - Specify output format(s).
    -o, --output                        - Write output to FILE (use '--output -' for stdout).
    --output-dir                        - Write project output to DIR (path is project relative)
    -M, --metadata                      - Metadata value (KEY:VALUE).
    --site-url                          - Override site-url for website or book output
    --execute                           - Execute code (--no-execute to skip execution).
    -P, --execute-param                 - Execution parameter (KEY:VALUE).
    --execute-params                    - YAML file with execution parameters.
    --execute-dir                       - Working directory for code execution.
    --execute-daemon                    - Keep Jupyter kernel alive (defaults to 300 seconds).
    --execute-daemon-restart            - Restart keepalive Jupyter kernel before render.
    --execute-debug                     - Show debug output for Jupyter kernel.
    --use-freezer                       - Force use of frozen computations for an incremental file render.
    --cache                             - Cache execution output (--no-cache to prevent cache).
    --cache-refresh                     - Force refresh of execution cache.
    --no-clean                          - Do not clean project output-dir prior to render
    --debug                             - Leave intermediate files in place after render.
    pandoc-args...                      - Additional pandoc command line arguments.
    --log                     <level>   - Path to log file
    --log-level               <level>   - Log level (info, warning, error, critical)
    --log-format              <format>  - Log format (plain, json-stream)
    --quiet                             - Suppress console output.
    --profile                           - Active project profile(s)
```

and in the `?quarto::quarto_render` that argument is `output_file`

**Reply by mcanouil** [reply]
What matters is the YAML frontmatter where "all" options are described in the reference pages of the documentation.
Quarto CLI has arguments to override the internal YAML values.
The "quarto" R package is just calling Quarto CLI and "-" is not allowed for object names in R.
